core/shared/src/main/scala/fs2/pipe.scala

```diff
@@ -75,6 +75,20 @@ object pipe {
   def delete[F[_],I](p: I => Boolean): Stream[F,I] => Stream[F,I] =
     _ pull { h => h.takeWhile((i:I) => !p(i)).flatMap(_.drop(1)).flatMap(_.echo) }
 
+  /**
+   * Emits only elements that are distinct from their immediate predecessors,
+   * using natural equality for comparison.
+   */
+  def distinctConsecutive[F[_],I]: Stream[F,I] => Stream[F,I] =
+    filterBy2((i1, i2) => i1 != i2)
+
+  /**
+   * Emits only elements that are distinct from their immediate predecessors
+   * according to `f`, using natural equality for comparison.
+   */
+  def distinctConsecutiveBy[F[_],I,I2](f: I => I2): Stream[F,I] => Stream[F,I] =
+    filterBy2((i1, i2) => f(i1) != f(i2))
```

Comment:
@pchiusano It's worse than that even -- `filterBy2` calls its predicate once *or twice* for each element depending on whether the chunk being processed needs any filtering. We could drop the optimization but it seems bad to rebuild each chunk pessimistically. I'll update the docs on `distinctConsecutiveBy` to make it clear that `f` should be cheap.